lec03-basic-r.Rmd

```diff
@@ -1,4 +1,732 @@
+classes.
+
+```{r}
+surveys[1, 1]   # first element in the first column of the data frame (as a vector)
+surveys[1, 6]   # first element in the 6th column (as a vector)
+surveys[, 1]    # first column in the data frame (as a vector)
+surveys[1]      # first column in the data frame (as a data.frame)
+surveys[1:3, 7] # first three elements in the 7th column (as a vector)
+surveys[3, ]    # the 3rd element for all columns (as a data.frame)
+head_surveys <- surveys[1:6, ] # equivalent to head(surveys)
+```
+
+`:` is a special function that creates numeric vectors of integers in increasing
+or decreasing order, test `1:10` and `10:1` for instance. This works similar to
+`seq`, which we looked at earlier in class:
+
+```{r}
+0:10
+seq(0, 10)
+
+# We can test if all elements are the same
+0:10 == seq(0,10)
+all(0:10 == seq(0,10))
+```
+
+You can also exclude certain parts of a data frame using the "`-`" sign:
+
+```{r}
+surveys[,-1]    # All columns, except the first
+surveys[-c(7:34786),] # Equivalent to head(surveys)
+```
+
+As well as using numeric values to subset a `data.frame` (or `matrix`), columns
+can be called by name, using one of the four following notations: <!-- Note sure how important it is to learn these difference vs just teaching the preferred way with the footnote that there are other ways also. -->
+
+```{r}
+head(surveys["species_id"])       # Result is a data.frame
+head(surveys[, "species_id"])     # Result is a vector
+head(surveys[["species_id"]])     # Result is a vector
+head(surveys$species_id)          # Result is a vector
+```
+
+For our purposes, the last three notations are equivalent. RStudio knows about
+the columns in your data frame, so you can take advantage of the autocompletion
+feature to get the full and correct column name.
+
+> ### Challenge
+>
+> 1. Create a `data.frame` (`surveys_200`) containing only the observations from
+>    row 200 of the `surveys` dataset.
+>
+> 2. Notice how `nrow()` gave you the number of rows in a `data.frame`?
+>
+>      * Use that number to pull out just that last row in the data frame.
+>      * Compare that with what you see as the last row using `tail()` to make
+>        sure it's meeting expectations.
+>      * Pull out that last row using `nrow()` instead of the row number.
+>      * Create a new data frame object (`surveys_last`) from that last row.
+>
+> 3. Use `nrow()` to extract the row that is in the middle of the data
+>    frame. Store the content of this row in an object named `surveys_middle`.
+>
+> 4. Combine `nrow()` with the `-` notation above to reproduce the behavior of
+>    `head(surveys)` keeping just the first through 6th rows of the surveys
+>    dataset.
+
+<!---
+```{r}
+## Answers
+surveys_200 <- surveys[200, ]
+surveys_last <- surveys[nrow(surveys), ]
+surveys_middle <- surveys[nrow(surveys)/2, ]
+surveys_head <- surveys[-c(7:nrow(surveys)),]
+```
+--->
+
+
+## Exploratory data analyses (EDA) in R
```

Comment:
I don't like how this header and the follow `###` headers look in the TOC on the HTML version. Can you just say `## R packages for data analysis` and change the `###` headers to `##`?